python/lsst/pipe/tasks/makeGen3SkyMap.py

```diff
@@ -0,0 +1,131 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import lsst.afw.geom as afwGeom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.daf.butler import Butler, DatasetType
+from lsst.skymap import skyMapRegistry
+
+
+class MakeGen3SkyMapConfig(pexConfig.Config):
+    """Config for MakeGen3SkyMapTask
+    """
+    datasetTypeName = pexConfig.Field(
+        doc="Name assigned to created skymap in butler registry",
+        dtype=str,
+        default="deepCoadd_skyMap",
+    )
+    name = pexConfig.Field(
+        doc="Name assigned to created skymap in butler registry",
+        dtype=str,
+        default=None,
+        optional=True
+    )
+    skyMap = skyMapRegistry.makeField(
+        doc="type of skyMap",
+        default="dodeca",
+    )
+    doWrite = pexConfig.Field(
+        doc="persist the skyMap? If False then run generates the sky map and returns it, "
+            "but does not save it to the data repository",
+        dtype=bool,
+        default=True,
+    )
+
+    def validate(self):
+        if self.name is None:
+            raise ValueError("The name field must be set to the name of the specific "
+                             "skymap to use when writing to the butler")
+
+
+class MakeGen3SkyMapTask(pipeBase.Task):
+    ConfigClass = MakeGen3SkyMapConfig
+    _DefaultName = "makeGen3SkyMap"
+
+    """This is a task to construct and optionally save a SkyMap into a gen3
+    butler repository.
+
+    Parameters
+    ----------
+    config : `MakeGen3SkyMapConfig` or None
+        Instance of a configuration class specifying task options, a default
+        config is created if value is None
+    butler : `str`
+        Path to a gen3 butler
+    collection : `str`
+        The name of the collection which the created skymap will be inserted
```

Comment:
I think it's probably better to accept a `Butler` instance in `run`, and actually construct it from a path and collection string in the `bin` script - I envision the shared logic for constructing a butler from command-line arguments to eventually move to common place, and when it does, that will reduce the need for changes to this task.